bin/compile

```diff
@@ -40,7 +40,18 @@ BUILD_DIR=$1
 CACHE_DIR=$2
 ENV_DIR=$3
 BAZEL_BUILD_PATH=$(cat ${ENV_DIR}/BAZEL_BUILD_PATH)_deploy.jar
+USE_BUILD_CACHE=$(cat ${ENV_DIR}/USE_BUILD_CACHE)
 
+echo "-----> BUILD_DIR: ${BUILD_DIR}"
+echo "-----> CACHE_DIR: ${CACHE_DIR}"
+echo "-----> ENV_DIR: ${ENV_DIR}"
+echo "-----> BAZEL_BUILD_PATH: ${BAZEL_BUILD_PATH}"
+echo "-----> USE_BUILD_CACHE: ${USE_BUILD_CACHE}"
```

Comment:
If the `BUILD_CACHE_BUCKET` variable is defined, we can assume the client intends to use cacheing.
Just minimizing the api surface area is good practice in general.

Consider just supporting this following 2 variables:
* `BUILD_CACHE_LOCATION`
* `BUILD_CACHE_GCP_CREDENTIALS`

You can check these individually and construct the final flags string the same way you do already.

Reply:
For example:

```bash
BUILD_CACHE_LOCATION=$(cat ${ENV_DIR}/BUILD_CACHE_LOCATION)
BUILD_CACHE_GCP_CREDENTIALS=$(cat ${ENV_DIR}/BUILD_CACHE_GCP_CREDENTIALS)

BAZEL_BUILD_FLAGS=""
if [[ -n $BUILD_CACHE_LOCATION ]]; then
  BAZEL_BUILD_FLAGS="$BAZEL_BUILD_FLAGS --remote_http_cache=${BUILD_CACHE_LOCATION} --remote_upload_local_results=false"
fi

if [[ -n $BUILD_CACHE_GCP_CREDENTIALS ]]; then
  BAZEL_BUILD_FLAGS="$BAZEL_BUILD_FLAGS --google_credentials=${BUILD_CACHE_GCP_CREDENTIALS}"
fi
```

This way clients could theoretically use this buildpack with no authentication as well